libpod/kube.go

```diff
@@ -72,7 +72,7 @@ func (p *Pod) GenerateForKube() (*v1.Pod, []v1.ServicePort, error) {
 			return nil, servicePorts, err
 		}
 		servicePorts = containerPortsToServicePorts(ports)
-		hostNetwork = p.config.InfraContainer.HostNetwork
+		hostNetwork = !p.config.InfraContainer.ContainerNetworkConfig.CreateNetNS
```

Comment:
Not accurate - also need to check whether the container has or lacks a network namespace

Reply:
`hostNetwork = (p.config.InfraContainer.NetMode == namespaces.NetworkMode(specgen.Host))` might work...

Reply:
Potentially. I think there's a NetMode function on the container - you could try that?